test cases/d/17 dub respects dub.selections.json/meson.build

```diff
@@ -0,0 +1,25 @@
+project('Dub dependency respects dub.selections.json', 'd')
+
+dub_exe = find_program('dub', required : false)
+if not dub_exe.found()
+    error('MESON_SKIP_TEST: Dub not found')
+endif
+
+dub_ver = dub_exe.version()
+if not dub_ver.version_compare('>=1.35.0')
+  error('MESON_SKIP_TEST: test requires dub >=1.35.0')
+endif
+
+dc = meson.get_compiler('d').cmd_array()[0]
+arch = host_machine.cpu_family()
+
+root = meson.source_root()
+run_command(dub_exe, 'build', '--deep', '--compiler', dc, '--arch', arch,
+            '--root', root,
+            check: true)
```

Comment:
It looks like basically all of these new test cases are running this command, which I think is supposed to recursively build dependencies, and then the tests maybe never actually build the dub project itself?

But don't we already have the built dependencies we need? If not, can we guarantee it is built in CI rather than potentially very slowly building dependencies on every CI run?

Reply:
> This adds five new test cases. Is it possible we can test multiple things in a single test case?

Yeah, I think some can be merged.

> It looks like basically all of these new test cases are running this command, which I think is supposed to recursively build dependencies, and then the tests maybe never actually build the dub project itself?
> 
> But don't we already have the built dependencies we need? If not, can we guarantee it is built in CI rather than potentially very slowly building dependencies on every CI run?

I did want to ask about this. I've written the commands in the meson files because I wanted to support people running the testsuite by themselves, without having to go through the CI script or reading test failures to find the line that tells them how to invoke dub to test dependencies. It is also way more manageable to keep track of which dependencies are needed / what versions / what compiler they require etc.

However, as you have said, this has the potential to build a lot of stuff during the tests but note that dub will properly (usually) cache the results so if they have been installed during CI they will not be rebuilt during the tests. This could be made a little better by having the new tests specify `"targetType": "none"` in their dub.json which will make the `dub build` command to only build the dependencies.

The usually aove refers to dub forcefully rebuilding stuff if a change in dependency occurs so:
- `dub build dep` (builds)


Now, if the issue is the potential of building  stuff outside of CI then yes, all the dependencies would need to be gathered and installed manually.

There is also another issue with installing them during the tests, there is the possibility of dub being called twice to build the same dependency and if it isn't present dub may get into a race-condition with the other process.